components/application-broker/internal/broker/provision.go

```diff
@@ -189,6 +196,33 @@ func (svc *ProvisionService) Provision(ctx context.Context, osbCtx osbContext, r
 	return resp, nil
 }
 
+// ProvisionProcess triggers provision process for other than broker (http) calls
+func (svc *ProvisionService) ProvisionProcess(req RestoreProvisionRequest) error {
+	var app *internal.Application
+	err := wait.Poll(500*time.Millisecond, 10*time.Second, func() (done bool, err error) {
+		app, err = svc.appSvcFinder.FindOneByServiceID(req.ApplicationServiceID)
+		if err != nil {
+			return false, errors.Wrap(err, "while finding application based on service ID")
```

Comment:
returning error will abort wait loop, is that intended? or you want to also retry on error?

Reply:
retry on error, fixed